aws/resource_aws_wafv2_regex_pattern_set.go

```diff
@@ -0,0 +1,267 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/wafv2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsWafv2RegexPatternSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafv2RegexPatternSetCreate,
+		Read:   resourceAwsWafv2RegexPatternSetRead,
+		Update: resourceAwsWafv2RegexPatternSetUpdate,
+		Delete: resourceAwsWafv2RegexPatternSetDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				idParts := strings.Split(d.Id(), "/")
+				if len(idParts) != 3 || idParts[0] == "" || idParts[1] == "" || idParts[2] == "" {
+					return nil, fmt.Errorf("Unexpected format of ID (%q), expected ID/NAME/SCOPE", d.Id())
+				}
+				id := idParts[0]
+				name := idParts[1]
+				scope := idParts[2]
+				d.SetId(id)
+				d.Set("name", name)
+				d.Set("scope", scope)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 256),
+			},
+			"lock_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"regular_expression_list": {
```

Comment:
how would we feel about updating this attribute to be a `Set ` with `Elem:  &schema.Schema{Type: schema.TypeString}`(I'm thinking a la `wafregional_regex_pattern_set`)? while the API does require a map formatting in a request and responds back with a map, we don't necessarily make use of the keys so I'm wondering if we (and the user) may benefit here by adopting a flatter schema type

Reply:
Yeah thought about it and wasn't sure what to do and stayed close to the API implementation as is done in general. Will change it.

Reply:
makes sense @pvanbuijtene.. ah before making any major changes, @bflad, what are your recommendations re: this schema definition?

Reply:
too late 😬 ... can role it back when needed

Reply:
We typically recommend following the API schema and naming where possible, even if its looks weird initially with single nested arguments in this case. We generally have to assume the API authors have further intentions of expanding the structures within the API, so then we'd later have to deprecate the simplified configurations to support the expanded structures. Much better to future-proof now where possible then deal with the work later. 👍 

Aside: Certainly curious there was an API choice of including the `List` suffix in the actual field name. Nothing wrong with following that usually, but `_list` could be debatable dropping from the Terraform attribute naming if can be confusing for practitioners writing configurations and expecting it to accept the `list` type (`= []`). This is why we can't have nice things. 😅

Reply:
The schema is back as it was, minus the `_list` post fix.